### ISSUE TITLE
bash completion for Mint's implementation of apt

### ISSUE THREAD

**Issue by ckristi** [linuxmint/linuxmint#16]
In Mint, the apt command is implemented as a python wrapper under /usr/local/bin/apt (part of the mint-system package).

However, the bash completion config for Mint's apt, seems to be missing completion for 'apt list' for example. Removing the /etc/bash_completion.d/apt-linux-mint (also from mint-system package) allows /usr/share/bash-completion/completions/apt to be taken into consideration for the apt command (and 'apt list --upgradable' autocompletes nicely), but it may break some other functionalities/commands provided by the wrapper. 

Could the two completion file configs be merged somehow?

Thank you,
Cristi

**Comment by mleo91** [comment]
I can confirm the bug (linux mint 19 cinnamon 64 bit beta). I had to add the "autoremove" tag to line 50 of the "apt-linux-mint" file so I have suggestions when I run the command in question with "apt".

It seems that some options are missing inside the wrapper; I also wonder if its disabling can result in a loss of functionality in the system.

Thanks for your attention,
Marco

**Comment by DarthFudd** [comment]
I'm getting the same problem in Mint 18. At a Bash command prompt I type "apt li", and then hit tab a few times looking for autocompletion to fill in "apt list", but there's no response. Similarly, if I type "apt list --upgr" and hit tab a few times, there's no autocompletion to "apt list --upgradable". I get the same behavior if I switch to an interactive sudo shell  with "sudo --login" or "sudo --shell".

I'm currently working around it by adding this line to my .bashrc file:

    source /usr/share/bash-completion/completions/apt

So far this seems to work with no problems, but I have NOT tested the workaround extensively for unwanted side-effects.

**Comment by dimonic** [comment]
Also not getting completions for git, similar workaround works (
```
source /usr/share/bash-completion/completions/git
```
).

**Comment by x10an14** [comment]
Still have this issue on Linux Mint 20.1

**Comment by matsonfamily** [comment]
yeah, i found that typing the bash built-in of **source** also worked (thanks to @DarthFudd above)... why in the heck was the really well written debian autocomplete (see the /usr/share/bash-completion/completions/ file) hacked up into a little tiny mint version?  and who edits a file like this without providing a date and name explanation at the top?  seems that someone just made a file to change the search keyword and then forgot to paste the rest in?  or maybe the author thought they were additive?

**Comment by smurphos** [comment]
> yeah

The file and it's commit history is here - https://github.com/linuxmint/mintsystem/blob/master/etc/bash_completion.d/apt-linux-mint

**Comment by hedenface** [comment]
Just to chime in, I'm on 20.1, and this works perfectly for me. Maybe it got fixed
